src/Compilers/CSharp/Portable/Symbols/Source/FieldSymbolWithAttributesAndModifiers.cs

```diff
@@ -0,0 +1,390 @@
+            if (attributesBag == null || !attributesBag.IsDecodedWellKnownAttributeDataComputed)
+            {
+                attributesBag = this.GetAttributesBag();
+            }
+
+            return (CommonFieldWellKnownAttributeData)attributesBag.DecodedWellKnownAttributeData;
+        }
+
+        /// <summary>
+        /// Returns data decoded from special early bound well-known attributes applied to the symbol or null if there are no applied attributes.
+        /// </summary>
+        /// <remarks>
+        /// Forces binding and decoding of attributes.
+        /// </remarks>
+        private CommonFieldEarlyWellKnownAttributeData GetEarlyDecodedWellKnownAttributeData()
+        {
+            var attributesBag = _lazyCustomAttributesBag;
+            if (attributesBag == null || !attributesBag.IsEarlyDecodedWellKnownAttributeDataComputed)
+            {
+                attributesBag = this.GetAttributesBag();
+            }
+
+            return (CommonFieldEarlyWellKnownAttributeData)attributesBag.EarlyDecodedWellKnownAttributeData;
+        }
+
+        internal sealed override CSharpAttributeData EarlyDecodeWellKnownAttribute(ref EarlyDecodeWellKnownAttributeArguments<EarlyWellKnownAttributeBinder, NamedTypeSymbol, AttributeSyntax, AttributeLocation> arguments)
+        {
+            CSharpAttributeData boundAttribute;
+            ObsoleteAttributeData obsoleteData;
+
+            if (EarlyDecodeDeprecatedOrExperimentalOrObsoleteAttribute(ref arguments, out boundAttribute, out obsoleteData))
+            {
+                if (obsoleteData != null)
+                {
+                    arguments.GetOrCreateData<CommonFieldEarlyWellKnownAttributeData>().ObsoleteAttributeData = obsoleteData;
+                }
+
+                return boundAttribute;
+            }
+
+            return base.EarlyDecodeWellKnownAttribute(ref arguments);
+        }
+
+        /// <summary>
+        /// Returns data decoded from Obsolete attribute or null if there is no Obsolete attribute.
+        /// This property returns ObsoleteAttributeData.Uninitialized if attribute arguments haven't been decoded yet.
+        /// </summary>
+        internal sealed override ObsoleteAttributeData ObsoleteAttributeData
+        {
+            get
+            {
+                var containingSourceType = (SourceMemberContainerTypeSymbol)ContainingType;
+                if (!containingSourceType.AnyMemberHasAttributes)
+                {
+                    return null;
+                }
+
+                var lazyCustomAttributesBag = _lazyCustomAttributesBag;
+                if (lazyCustomAttributesBag != null && lazyCustomAttributesBag.IsEarlyDecodedWellKnownAttributeDataComputed)
+                {
+                    var data = (CommonFieldEarlyWellKnownAttributeData)lazyCustomAttributesBag.EarlyDecodedWellKnownAttributeData;
+                    return data != null ? data.ObsoleteAttributeData : null;
+                }
```

Comment:
Could the `if` be replaced with `var data = GetEarlyDecodedWellKnownAttributeData();`?